ledger/src/blockstore_db.rs

```diff
@@ -403,9 +414,9 @@ impl Rocks {
         // this is only needed for LedgerCleanupService. so guard with PrimaryOnly (i.e. running solana-validator)
         if matches!(access_type, AccessType::PrimaryOnly) {
             for cf_name in cf_names {
-                // this special column family must be excluded from LedgerCleanupService's rocksdb
+                // these special column families must be excluded from LedgerCleanupService's rocksdb
                 // compactions
-                if cf_name == TransactionStatusIndex::NAME {
+                if cf_name == TransactionStatusIndex::NAME || cf_name == ProgramCosts::NAME {
```

Comment:
Nit: could we factor these exceptions into a separate structure above in case we need to add more in the future (similar to `cf` above). Maybe something like:

```
let no_compaction_filter_cfs: HashSet<&'static str>  = vec![TransactionStatusIndex::NAME, ProgramCosts::NAME].into_iter().collect()
```

Reply:
Since we have two places doing same thing to exclude same CFs from compaction. I added a function to share: 
```
fn excludes_from_compaction( cf_name: &str) -> bool {
    let no_compaction_cfs: HashSet<&'static str> = vec![
        columns::TransactionStatusIndex::NAME,
        columns::ProgramCosts::NAME,
    ].into_iter().collect();

    no_compaction_cfs.get(cf_name).is_some()
}
```